libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -489,16 +489,13 @@ void CodeTransform::operator()(FunctionDefinition const& _function)
 
 	m_assembly.setSourceLocation(_function.location);
 	int stackHeightBefore = m_assembly.stackHeight();
-	AbstractAssembly::LabelID afterFunction = m_assembly.newLabelId();
 
```

Comment:
It might be best to implement a `setStackHeight` function in `AbstractAssembly` and call it here with `m_assembly.setStackHeight(height)`. At the end of the function, call it again with `m_assembly.setStackHeight(stackHeightBefore)`.

Reply:
The function will just set `m_stackHeight` in `EVMAssembly` (and other derived classes)?

I can do that, but I have no idea what is happening 😅

Reply:
I think I got it now - ``setStackHeight`` has to be called *after* the ``if`` in the next line - in the ``m_evm15 == true`` case ``appendBeginSub`` will itself adjust the stack for the function arguments - but that's also done with ``height``, so currently that's done twice. So just moving ``m_assembly.setStackHeight(height)`` one statement down past the ``if`` should be all that's needed.

Reply:
Note that this is exactly the difference between ``-stackHeightBefore`` and ``-stackHeightBefore + height`` we had before this PR in the ``if(m_evm15)``.